app/src/main/java/com/devshish/internship/domain/repository/IAuthRepository.kt

```diff
@@ -0,0 +1,11 @@
+package com.devshish.internship.domain.repository
+
+import android.net.Uri
+import com.devshish.internship.data.model.TokenDTO
+
+interface IAuthRepository {
+
+    val authRequest: Uri
+
+    suspend fun getToken(code: String): TokenDTO
+}
```

Comment:
Also , here I suggest to create UseCase

so it will hide some logic , not only data
<img width="392" alt="image" src="https://user-images.githubusercontent.com/23121846/135872534-b54fd8e9-ab37-4015-84d5-63f1bf2c6d57.png">

And in implementation inject TokenRepository, and store token inside of UseCase implementation